src/Main.java

```diff
@@ -0,0 +1,163 @@
+				break;
+			case 3:
+				nums = CircleSort.Sort(nums);
+				break;
+			case 4:
+				CocktailSort cs = new CocktailSort();
+				cs.sort(nums);
+				break;
+			case 5:
+				InsertionSort is = new InsertionSort();
+				is.sort(nums);
+				break;
+			case 6:
+				CycleSort.cycleSort(nums, nums.length);
+				break;
+			case 7:
+				GnomeSort.gnomeSort(nums);
+				break;
+			case 8:
+				HeapSort.sort(nums);
+				break;
+			case 9:
+				PigeonHoleSort.pigeonholeSort(nums, nums.length);
+				break;
+			case 10:
+				QuickSort qs = new QuickSort();
+				qs.sort(nums, 0, nums.length - 1);
+				break;
+			case 11:
+				RadixSort.radixSort(nums);
+				break;
+			case 12:
+				SelectionSort.sort(nums);
+				break;
+			case 13:
+				ShakerSort.sort(nums);
+				break;
+			case 14:
+				ShellSort.sort(nums);
+				break;
+			case 15:
+				SlowSort.slowSort(nums, 0, nums.length -1);
+				break;
+			case 16:
+				nums = StoogeSort.stoogeSort(nums, 0, nums.length-1);
+				break;
+			}
+			printNumbers(nums);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private static void printNumbers(int[] arr) {
+		System.out.println("Sorted:");
+		for (int i = 0; i < arr.length; ++i) {
+			System.out.println(arr[i]);
+		}
+	}
+
+	private static void printOptions() {
+		System.out.println("Enter a number to select an algorithm");
+		System.out.println("1 - Bead");
+		System.out.println("2 - Bubble");
+		System.out.println("3 - Circle");
+		System.out.println("4 - Cocktail");
+		System.out.println("5 - Insert");
+		System.out.println("6 - Cycle");
+		System.out.println("7 - Gnome");
+		System.out.println("8 - Heap");
+		System.out.println("9 - Pigeonhole");
+		System.out.println("10 - Quick");
+		System.out.println("11 - Radix");
+		System.out.println("12 - Selection");
+		System.out.println("13 - Shaker");
+		System.out.println("14- Shell");
+		System.out.println("15 - Slow");
+		System.out.println("16 - Stooge");
+	}
+
+	private static void printInstructions() {
+		System.out.println("Enter the numbers you want to sort, separated by enters");
+		System.out.println("Enter an empty line to start sorting");
+	}
+
+	private static Integer readAlg() {
+		Scanner in = new Scanner(System.in);
+		String s = in.nextLine();
+
+		try {
+			Integer number = Integer.parseInt(s);
+			return number;
+		} catch (NumberFormatException nfe) {
+			System.out.println("Please add a number\n");
+			throw nfe;
+		}
+
+	}
+
+	private static int[] readNumbers() {
+		ArrayList<Integer> array = new ArrayList<Integer>();
+		Scanner in = new Scanner(System.in);
+
+		while (in.hasNextLine()) {
+			try {
+				String s = in.nextLine();
+				if (s.equals("")) {
+					break;
+				}
+				Integer number = Integer.parseInt(s);
+				array.add(number);
+			} catch (NumberFormatException nfe) {
+				System.out.println("Please enter a number");
+			}
+		}
+
+		int[] res = new int[array.size()];
+		for (int i = 0; i < array.size(); i++) {
+			res[i] = array.get(i);
+		}
```

Comment:
return array.toArray(); 
It would be a bit more readable

Reply:
Variable name shouldn't be array as its type is ArrayList and not array

Reply:
Great observations, I will correct them